code/Controller/AssetAdmin.php

```diff
@@ -273,6 +323,195 @@ public function getClientConfig()
+                continue;
+            }
+            if (!$file->canView()) {
+                $this->jsonError(403);
+            }
+            $fileIDs[] = $file->ID;
+        }
+        if (count($fileIDs) > 0) {
+            $liveFiles = Versioned::get_by_stage(File::class, Versioned::LIVE)
+                ->filter(['ID' => $fileIDs]);
+            /** @var Versioned|RecursivePublishable|File $liveFile */
+            foreach ($liveFiles as $liveFile) {
+                $count = $liveFile->findOwners(false)->count();
+                // This translation key is from a class that has since been removed
+                $class = 'SilverStripe\\AssetAdmin\\GraphQL\\Resolvers\\PublicationResolver';
+                $data[] = [
+                    'id' => $liveFile->ID,
+                    'count' => $count,
+                    'message' => _t(
+                        "$class.OWNER_WARNING",
+                        'File "{file}" is used in {count} place|File "{file}" is used in {count} places.',
+                        [
+                            'file' => $liveFile->Title,
+                            'count' => $count
+                        ]
+                    ),
+                ];
+            }
+        }
+        $this->extend('updateApiReadLiveOwnerCounts', $data, $request);
+        return $this->jsonSuccess(200, $data);
+    }
+
+    public function apiReadUsage(HTTPRequest $request): HTTPResponse
+    {
+        $id = (int) $request->param('ID');
+        /* @var Folder&AssetAdminFile $file */
+        $file = $this->getFileByID($id);
+        if (!$file->canView()) {
+            $this->jsonError(403);
+        }
+        $count = $file->getFilesInUse()->count();
+        $data = ['count' => $count];
+        $this->extend('updateApiReadUsage', $data, $request);
+        return $this->jsonSuccess(200, $data);
+    }
+
+    public function apiArchive(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $ids = $this->getPostedJsonValue($request, 'ids');
+        if (!is_array($ids) || empty($ids)) {
+            $this->jsonError(400);
+        }
+        $files = File::get()->filter(['ID' => $ids]);
+        if ($files->count() !== count($ids)) {
+            $this->jsonError(400);
+        }
+        foreach ($files as $file) {
+            if (!$file->canDelete()) {
+                $this->jsonError(403);
+            }
+        }
+        foreach ($files as $file) {
+            if ($file->has_extension(Versioned::class)) {
+                $file->doArchive();
+            } else {
+                $file->delete();
+            }
+        }
+        return $this->jsonSuccess(204);
+    }
+
+    public function apiMove(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $ids = $this->getPostedJsonValue($request, 'ids');
+        $folderID = (int) $this->getPostedJsonValue($request, 'folderID');
```

Comment:
```suggestion
        $folderID = $this->getPostedJsonValue($request, 'folderID');
```
We don't need to cast it, PHP will do that for us when relevant.

Reply:
Updated